test/UserDbSpec.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeOperators              #-}
+
+module UserDbSpec where
+
+import           Test.Hspec
+import           Test.QuickCheck
+
+import           Control.Monad.Except      (runExceptT)
+import           Control.Monad.Reader      (runReaderT)
+
+import           Servant
+import           Database.Persist.Types      (Filter)
+import           Database.Persist.Sql        (ConnectionPool, transactionUndo)
+import           Database.Persist.Postgresql (Entity (..), fromSqlKey, insert,
+                                              selectFirst, selectList, deleteWhere,
+                                              (==.), runSqlPool)
+
+import           Config (App (..), Config (..), Environment (..), makePool)
+import           Models
+import           Api.User
+
+
+runAppToIO :: Config -> App a -> IO a
+runAppToIO config app = do
+  result <- runExceptT $ runReaderT (runApp app) config
+  case result of
+    Left err -> error $ show err
```

Comment:
Additionally, you'll sometimes want to make assertions about whether or not the response was a `Left ServantErr`. So this might make more sense as simply

```haskell
runAppIO :: Config -> App a -> IO (Either ServantErr a)
runAppIO cfg = runExceptT . flip runReaderT cfg . runApp
```